woocommerce/woocommerce-ios
# Set the right timezone when setting the expire date of a Coupon

Closes #6687 

### Description
In this PR I took care of the timezone in the expiration date field for Coupon during the editing/addition phase.
- Now the `updateCoupon` and `createCoupon` methods accept the timezone parameter because we are setting the site timezone for updating the `date_expires` fields.
- The `CouponExpiryDateView` now works by default with the site timezone.

### Testing instructions
- Open a Coupon detail, and start the editing flow.
- Try to edit the date of the coupon, and save it.
- Launch Wormholy from Settings, and inspect the API request made.
- You should see in the `date_expires` field the time configured on the app at the time `00:00:00`.
- You should see in the `date_expires_gmt` field the date GMT configured on the app (you could also check the timezone configured on your website by navigating on wp-admin > Settings > General > TimeZone to see if the date is correct).


---
